scripts/bootstrap.sh

```diff
@@ -43,34 +81,47 @@ install_nix() {
 			--extra-conf "trusted-users = root @admin" \
 			--ssl-cert-file "$NIX_FINAL_SSL_FILE"
 	echo "=== nix installed..."
+	fi
 
+	if pgrep -q nix-daemon;then 
+		echo "=== ✅ nix daemon running, doing nothing"
+	else
 	echo "=== sourcing nix daemon so we can use it in this script..."
 	export NIX_SSL_CERT_FILE="$NIX_FINAL_SSL_FILE"
 	# shellcheck source=/dev/null
 	. /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
 	echo "=== nix daemon sourced..."
+	fi
 }
 
 install_devbox() {
-	echo "=== installing devbox..."
-	curl -fsSL https://get.jetpack.io/devbox | FORCE=1 bash
-	echo "=== devbox installed..."
+	if command -v devbox version >/dev/null 2>&1; then
+		echo "=== ✅ devbox is already installed, doing nothing"
+	else
+		echo "=== installing devbox..."
+		curl -fsSL https://get.jetpack.io/devbox | FORCE=1 bash
+		echo "=== devbox installed..."
+	fi
 }
 
 add_current_user_to_admin_group() {
+	if id -Gn "$USER" | grep -q -w admin;
+	then
+		echo "=== ✅ user is already added to admin group, doing nothing"
+	else
 	echo "=== add current user to admin group"
 	sudo dseditgroup -o edit -a "$(whoami)" -t user admin
+	fi
 }
 
 install_direnv() {
 	if command -v direnv >/dev/null 2>&1; then
-		echo "=== direnv is already installed, doing nothing"
-		DID_INSTALL_DIRENV=0
+		echo "=== ✅ direnv is already installed, doing nothing"
+		DIDNT_INSTALL_DIRENV=1
 	else
 		echo "=== direnv is not installed, installing..."
 		nix profile install nixpkgs#direnv
```

Comment:
This step fails:
```
elliott.hilaire@ehilaire-4YDHJ2 scripts % ./bootstrap.sh                                                                                                                    
=== ✅ user is already added to admin group, doing nothing
=== 🔐 generating combined CA certificate from system keychain...
=== 🔐 combined CA certificate generated
=== ✅ Netskope certificate is already placed, doing nothing
=== installing nix (requires sudo)...
info: downloading installer https://install.determinate.systems/nix/tag/v0.17.1/nix-installer-aarch64-darwin
`nix-installer` needs to run as `root`, attempting to escalate now via `sudo`...
 INFO Step: Create an encrypted APFS volume `Nix Store` for Nix on `disk3` and add it to `/etc/fstab` mounting on `/nix`
 INFO Step: Provision Nix
 INFO Step: Create build users (UID 301-332) and group (GID 30000)
 INFO Step: Configure Time Machine exclusions
 INFO Step: Configure Nix
 INFO Step: Configuring zsh to support using Nix in non-interactive shells
 INFO Step: Unload, then recreate a `launchctl` plist to put Nix into your PATH
 INFO Step: Configure Nix daemon related settings with launchctl
 INFO Step: Remove directory `/nix/temp-install-dir`
Nix was installed successfully!
To get started using Nix, open a new shell or run `. /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh`

=== nix installed...
=== ✅ nix daemon running, doing nothing
=== ✅ devbox is already installed, doing nothing
=== direnv is not installed, installing...
./bootstrap.sh: line 123: nix: command not found
```
Even though nix is installed its not found in via PATH yet. 

Opening a new shell and re-running bootstrap.sh works.
```
elliott.hilaire@ehilaire-4YDHJ2 scripts % ./bootstrap.sh
=== ✅ user is already added to admin group, doing nothing
=== 🔐 generating combined CA certificate from system keychain...
=== 🔐 combined CA certificate generated
=== ✅ Netskope certificate is already placed, doing nothing
=== ✅ nix is already installed, checking ssl certificate
warning: 'show-config' is a deprecated alias for 'config show'
=== ✅ nix ssl certificate pointing to correct path
=== ✅ nix daemon running, doing nothing
=== ✅ devbox is already installed, doing nothing
=== direnv is not installed, installing...
=== direnv installed
=== installing nix-direnv...
=== nix-direnv installed
=== direnvrc doesn't exist, creating it with config
================================================================
Nix, direnv, and devbox have been installed and setup
direnv setup will only be activated if you start a new shell session (e.g. open a new tab on your terminal)
================================================================

If you've had any issues with this install process please reach out to #team_delivery_eng on slack
elliott.hilaire@ehilaire-4YDHJ2 scripts % 
```

Looking at the changes it appears this might be a pre-existing bug.

Reply:
I think this is an introduced bug. I conditionally source the nix-daemon based on if it is running. The nix-daemon seems to be running, but the issue then is as you point out, the terminal's environment has not been reloaded and nix is not on path.

Reply:
I've decided the best path forward is to always source the daemon. Seems to me like there is no harm sourcing in any of the possible contexts.